p2p/net/swarm/dial_test.go

```diff
@@ -552,7 +538,7 @@ func newSilentListener(t *testing.T) ([]ma.Multiaddr, net.Listener) {
 }
 
 func TestDialSimultaneousJoin(t *testing.T) {
-	const dialTimeout = 250 * time.Millisecond
+	const dialTimeout = 3 * time.Second
```

Comment:
Is this necessary? This one test now takes 3s.

Reply:
Yes. I'm pretty sure I saw this test flake here because of the dialTimeout. 1s might be okay as well but 3s definitely is.

Reply:
Fine for now, but long term, we'll need to find a faster solution for this.

Reply:
There's much much lower hanging fruit than this. 

1. We can use a build cache
2. We can split the 32bit tests/race tests into a separate job so they can run in parallel. With a build cache, these won't waste too much time in duplicated efforts.